src/main/java/ir/sahab/dockercomposer/DockerComposeRunner.java

```diff
@@ -106,35 +106,46 @@ public void stopService(String serviceName) {
         executeDockerCompose("stop", serviceName);
     }
 
-    private Service createService(String name) {
-        String serviceInfo = getContainerInfo(name, projectName);
+    private Service createService(String serviceName) {
+        String serviceInfo = getContainerNameAndPorts(serviceName, projectName);
         if (StringUtils.isEmpty(serviceInfo)) {
-            throw new IllegalStateException("Unexpected output of docker ps for service " + name
+            throw new IllegalStateException("Unexpected output of docker ps for service " + serviceName
                     + ". This is mostly due to your container is terminated early.");
         }
 
-        return parseServiceInfo(serviceInfo, name);
+        return createService(serviceInfo, serviceName);
     }
 
     /**
-     * Parses the docker-compose service state line and creates a service from the parsed result.
+     * Parses the 'docker ps' state line and creates a service from the parsed result.
      */
-    // We might later convert this to an strategy based on docker-compose version
-    private Service parseServiceInfo(String serviceInfo, String name) {
+    private Service createService(String serviceInfo, String serviceName) {
         /* Service info is like this:
          * zookeeper-test-1#2888/tcp, 3888/tcp, 0.0.0.0:32768->2181/tcp
          * We want to extract these: zookeeper-test-1 as id and 2181 as the default port which
          * is mapped to port 32768. The service externalIp is 0.0.0.0 which means local host.
          */
-        final Pattern portPattern
-                = Pattern.compile("((\\d+).(\\d+).(\\d+).(\\d+)):(\\d+)->(\\d+)/tcp");
+        final Pattern portPattern = Pattern.compile("((\\d+).(\\d+).(\\d+).(\\d+)):(\\d+)->(\\d+)/tcp");
         final int externalPort = 6;
         final int internalPort = 7;
         final String externalIp = "127.0.0.1";
 
         String[] serviceInfoParts = serviceInfo.split("#");
+        if (serviceInfoParts.length < 1 || serviceInfoParts.length > 2) {
+            throw new IllegalArgumentException("Invalid service info: " + serviceInfo);
+        }
+
+        // Extract container's internal IP and add service name to internal IP resolution into Java DNS.
         String id = serviceInfoParts[0];
+        String internalIp;
+        try {
+            internalIp = getContainerIp(id);
+            NameServiceEditor.addHostnameToNameService(serviceName, internalIp);
+        } catch (Exception e) {
+            throw new IllegalStateException("Could not set mapping of service name and internal IP", e);
+        }
 
+        // Extract port mapping if defined for the container
         Map<Integer, Integer> portMappings = new HashMap<>();
         if (serviceInfoParts.length == 2) {
```

Comment:
Remove this 'if' statement. You have checked it and now you are sure the length is 2.

Reply:
We've already checked the length is one or two. If port mapping is not defined for the container list will have one member.